api/v1beta1/common_types.go

```diff
@@ -29,6 +29,9 @@ type KubernetesConfig struct {
 	ExistingPasswordSecret *ExistingPasswordSecret          `json:"redisSecret,omitempty"`
 	ImagePullSecrets       *[]corev1.LocalObjectReference   `json:"imagePullSecrets,omitempty"`
 	UpdateStrategy         appsv1.StatefulSetUpdateStrategy `json:"updateStrategy,omitempty"`
+	// +kubebuilder:validation:Enum=LoadBalancer;NodePort;ClusterIP
+	ServiceType        string            `json:"serviceType,omitempty"`
```

Comment:
instead of two blocks, can we convert this into an object like this:

```yaml
service:
  type: NodePort
  annotations:
    foo: bar
```

Reply:
You are right，I've been reckless.